### PR TITLE
Fix issue 3577

### LINKED ISSUE #3577: Add Ubuntu 24.04 to Vagrant Playbook Check Job
ref: #3501

The Vagrant playbook check job needs expanding to cover Ubuntu 24.04.

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes #3577 

Adds Ubuntu 24.04 support to the Vagrant playbook check process.

Required changes to compiling NASM on Ubuntu 24.04
Also a change to the VPC build tests to build using the devkit, rather than gcc 7.5.

##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [X] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)

### CODE REVIEW

**Comment by steelhead31** [comment]
:heavy_check_mark:  VPC / Ubuntu 24 / JDK21 / https://ci.adoptium.net/job/SFR-VPC-AZURE-POC/OS=Ubuntu2404,label=vagrant/72/

**Review by Haroon-Khel — APPROVED** [review]

File: ansible/pbTestScripts/buildJDK.sh [hunk 1]
```diff
@@ -165,14 +163,19 @@ if [[ "$(uname -m)" == "armv7l" && "$VARIANT" == "openj9" ]]; then
...
-	findJDK7	
+	findJDK7
 fi
 
 # Don't build the debug-images as it takes too much space, and doesn't benefit VPC
 # See: https://github.com/adoptium/infrastructure/issues/2033
 export CONFIGURE_ARGS="--with-native-debug-symbols=none"
 export BUILD_ARGS="--custom-cacerts false"
 
+# For Ubutu24.04 Support - Dont Use gcc-7
```
Comment (sxa) [inline ansible/pbTestScripts/buildJDK.sh:174]:
```suggestion
# For Ubuntu24.04 Support - Don't Use gcc-7
```

**Review by sxa — CHANGES_REQUESTED** [review]
One line comment fix :-)

What is the history of the nasm patch? I was initally concerned about provenance / SBoM content but nasm is only used for OpenJ9 AFAIK. Is there a newer version of nasm that has the fix needed here? I guess our other option would be to skip it on Ubuntu 24.04 ...

**Comment by pshipton** [comment]
OpenJ9 isn't planning to build on Ubuntu 24_04, although I suppose a developer might want to, as long as they don't want to run the resulting JVM elsewhere. I don't see an issue installing a newer nasm version on this platform. The OpenJ9 build instructions specify v2.13.03 or newer. There are some other OpenJ9 issues trying to compile on 24_04 atm, although they should be fixed within the next week or so.
https://github.com/eclipse-openj9/openj9/issues/20170
https://github.com/eclipse-openj9/openj9/issues/20171

**Comment by steelhead31** [comment]
> OpenJ9 isn't planning to build on Ubuntu 24\_04, although I suppose a developer might want to, as long as they don't want to run the resulting JVM elsewhere. I don't see an issue installing a newer nasm version on this platform. The OpenJ9 build instructions specify v2.13.03 or newer. There are some other OpenJ9 issues trying to compile on 24\_04 atm, although they should be fixed within the next week or so. [eclipse-openj9/openj9#20170](https://github.com/eclipse-openj9/openj9/issues/20170) [eclipse-openj9/openj9#20171](https://github.com/eclipse-openj9/openj9/issues/20171)

Thanks @pshipton , I'll bump NASM to the current stable release 2.16.03 :)

**Review by karianna — APPROVED** [review]

File: ansible/pbTestScripts/buildJDK.sh [hunk 2]
```diff
@@ -165,14 +163,19 @@ if [[ "$(uname -m)" == "armv7l" && "$VARIANT" == "openj9" ]]; then
...
 fi
 
 # Don't build the debug-images as it takes too much space, and doesn't benefit VPC
 # See: https://github.com/adoptium/infrastructure/issues/2033
 export CONFIGURE_ARGS="--with-native-debug-symbols=none"
 export BUILD_ARGS="--custom-cacerts false"
 
+# For Ubuntu24.04 Support - Don't Use gcc-7
+if grep 'noble' /etc/*-release >/dev/null 2>&1; then
+export BUILD_ARGS="--custom-cacerts false --use-adoptium-devkit gcc-11.3.0-Centos7.9.2009-b03"
```
Comment (karianna) [inline ansible/pbTestScripts/buildJDK.sh:176]:
```suggestion
        export BUILD_ARGS="--custom-cacerts false --use-adoptium-devkit gcc-11.3.0-Centos7.9.2009-b03"
```

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/nasm/tasks/main.yml [hunk 3]
```diff
@@ -71,43 +71,51 @@
...
+  shell: cd /tmp && wget https://www.nasm.us/pub/nasm/releasebuilds/2.16.03/nasm-2.16.03.tar.gz
   when:
-    - (nasm_installed.rc is defined) and ((nasm_installed.rc != 0 ) or (nasm_installed.rc == 0 and nasm.stdout is version_compare('2.13.03', operator='ne')) )
+    - (nasm_installed.rc is defined) and ((nasm_installed.rc != 0 ) or (nasm_installed.rc == 0 and nasm.stdout is version_compare('2.16.03', operator='ne')) )
   tags: nasm
 
 - name: Extract nasm sources
   unarchive:
-    src: /tmp/nasm-2.13.03.tar.gz
+    src: /tmp/nasm-2.16.03.tar.gz
```
Comment (karianna) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/nasm/tasks/main.yml:81]: Feels like the version could get extracted.

Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/nasm/tasks/main.yml]: Updated. Thanks!

**Review by karianna — APPROVED** [review]

**Pull request merged**